app/models/authenticated_user.rb

```diff
@@ -0,0 +1,11 @@
+class AuthenticatedUser < User
+  def payload
+    attrs = {
```

Comment:
I think you can call `super` here to avoid duplicating the original functionality and just add the "mfa" attribute.

Something like:

```ruby
def payload
  super.merge({"mfa" => mfa_level})
end
```

Also I think it would be friendly to leave some comment at the top of the class (or method) explaining the purpose.